src/main/java/com/redhat/devtools/intellij/knative/actions/func/BuildAction.java

```diff
@@ -191,22 +175,27 @@ protected void doExecuteAction(Project project, Function function, String regist
         }
     }
 
-    protected void doExecute(FuncActionTask task, Kn knCli, String namespace, String localPathFunc, String registry, String image) throws IOException {
+    protected void doExecute(FuncActionTask task, Kn knCli, String namespace, String localPathFunc, ImageRegistryModel model) throws IOException {
         ConsoleView terminalExecutionConsole = task != null ? task.getTerminalExecutionConsole() : null;
         java.util.function.Function<ProcessHandlerInput, ExecProcessHandler> processHandlerFunction = task != null ? task.getProcessHandlerFunction() : null;
         ProcessListener processListener = task != null ? task.getProcessListener() : null;
-        knCli.buildFunc(localPathFunc, registry, image, terminalExecutionConsole, processHandlerFunction, processListener);
+        knCli.buildFunc(localPathFunc, model, terminalExecutionConsole, processHandlerFunction, processListener);
     }
 
     protected boolean isActionConfirmed(Project project, String name, String funcNamespace, String activeNamespace) {
         return true;
     }
 
-    protected String getImageFromUser(String name) {
+    protected ImageRegistryModel getImageFromUser(String name) {
         String defaultUsername = System.getProperty("user.name");
         String defaultImage = "quay.io/" + defaultUsername + "/" + name + ":latest";
-        Messages.InputDialog dialog = new Messages.InputDialog(null, "Provide full image name in the form [registry]/[namespace]/[name]:[tag] (e.g quay.io/boson/image:latest)",
-                "Build Function " + name, null, defaultImage,
+        InputDialogWithCheckbox dialog = new InputDialogWithCheckbox("Provide full image name in the form [registry]/[namespace]/[name]:[tag] (e.g quay.io/boson/image:latest)",
```

Comment:
[] are generally used to denote optional parts so maybe should be `registry/namespace/name[:tag]`